src/lib/client-cloud-services/AWSStorageService.js

```diff
@@ -0,0 +1,475 @@
+/**
+ * @file        - AWS Storage Service
+ * @exports     - `AWSStorageService`
+ * @since       - 5.0.1
+ * @version     - 1.0.0
+ * @implements  - BaseStorageService
+ * @see {@link https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html | X-Amz-Credential}
+ * @see {@link https://docs.aws.amazon.com/directconnect/latest/APIReference/CommonParameters.html#CommonParameters-X-Amz-Credential | X-Amz-Credential}
+ */
```

Comment:
TODO  - We can publish this cloud service as npm module and use it across all the repositories